packages/node/src/integrations/localvariables/common.ts

```diff
@@ -0,0 +1,118 @@
+import type { StackFrame, StackParser } from '@sentry/types';
+import type { Debugger } from 'inspector';
+
+export type Variables = Record<string, unknown>;
+
+export type RateLimitIncrement = () => void;
+
+/**
+ * Creates a rate limiter
+ * @param maxPerSecond Maximum number of calls per second
+ * @param enable Callback to enable capture
+ * @param disable Callback to disable capture
+ * @returns A function to call to increment the rate limiter count
+ */
+export function createRateLimiter(
+  maxPerSecond: number,
+  enable: () => void,
+  disable: (seconds: number) => void,
+): RateLimitIncrement {
+  let count = 0;
+  let retrySeconds = 5;
+  let disabledTimeout = 0;
+
+  setInterval(() => {
+    if (disabledTimeout === 0) {
+      if (count > maxPerSecond) {
+        retrySeconds *= 2;
+        disable(retrySeconds);
+
+        // Cap at one day
+        if (retrySeconds > 86400) {
+          retrySeconds = 86400;
+        }
+        disabledTimeout = retrySeconds;
+      }
+    } else {
+      disabledTimeout -= 1;
+
+      if (disabledTimeout === 0) {
+        enable();
+      }
+    }
+
+    count = 0;
+  }, 1_000).unref();
+
+  return () => {
+    count += 1;
+  };
+}
+
+// Add types for the exception event data
+export type PausedExceptionEvent = Debugger.PausedEventDataType & {
+  data: {
+    // This contains error.stack
+    description: string;
+  };
+};
+
+/** Could this be an anonymous function? */
+export function isAnonymous(name: string | undefined): boolean {
+  return name !== undefined && ['', '?', '<anonymous>'].includes(name);
```

Comment:
Since this function is called a lot, we should avoid creating unnecessary objects/arrays, since it will create GC bottlenecks.

```suggestion
  return name !== undefined && (name.length === 0 || name === '?' || name === '<anonymous>')
```